index.js

```diff
@@ -7,9 +8,10 @@ module.exports = async function({retry} = {}, {pkg, npm, options}, cb) {
   npmlog.level = npm.loglevel || 'warn';
   const client = new RegClient({log: npmlog, retry});
   try {
-    const data = await promisify(client.get.bind(client))(`${npm.registry}${pkg.name.replace('/', '%2F')}`, {
-      auth: npm.auth,
-    });
+    const data = await promisify(client.get.bind(client))(
+      new URL(pkg.name.replace('/', '%2F'), npm.registry).toString(),
+      {auth: npm.auth}
+    );
```

Comment:
I think a better way to do it is [`url.resolve()`](https://nodejs.org/api/url.html#url_url_resolve_from_to)

```
const {resolve} = require('url')
resolve(npm.registry, pkg.name.replace('/', '%2F'))
```

Reply:
I think so to, but `resolve` is part of what Node API doc describe as 'Legacy URL API':  https://nodejs.org/api/url.html#url_legacy_url_api

Does that mean it's deprecated ? Legacy is ambiguous...

Reply:
hmm my understanding is that `require('url').Url` is the legacy URL API and that it’s returned by `require('url').parse`. resolve is not a legacy API

Reply:
ahhhh... I get it now! I made the change as suggested !